contracts/auth/MultisigBase.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+
+import { IMultisigBase } from '../interfaces/IMultisigBase.sol';
+
+contract MultisigBase is IMultisigBase {
+    struct Voting {
+        uint256 voteCount;
+        mapping(address => bool) hasVoted;
+    }
+
+    struct Signers {
+        address[] accounts;
+        uint256 threshold;
+        mapping(address => bool) isSigner;
+        // bytes32 is for vote topic hash
+        mapping(bytes32 => Voting) voting;
+    }
+
+    uint256 public signerEpoch;
+    mapping(uint256 => Signers) public signersPerEpoch;
```

Comment:
Epoch is only needed for domain separation for vote tracking. So, I think we should remove accounts, threshold, isSigner into a separate struct that can be overwritten instead of being epoch separated. This should save some gas too I imagine

Reply:
A benefit of this is that queries for the current signer set don't require epoch to be passed in.

Gas savings matter less, since this code isn't triggered too often

Reply:
You can't overwrite struct in solidity. That's why it needs to be epoch separated